### ISSUE TITLE
Build failures with GHC-7.10.3 and GHC-7.8.4

### ISSUE THREAD

**Issue by phadej** [mgsloan/store#142]
```
src/Data/Store/Impl.hs:302:1:
    parse error (possibly incorrect indentation or mismatched brackets)
```

this is due

```haskell
type family TypeErrorMessage (a :: Symbol) :: Constraint where
#if MIN_VERSION_base(4,9,0)
    TypeErrorMessage a = TypeError (Text a)
#endif

-- Similarly to splitting up the generic class into multiple classes, we
-- also split up the one for sum types.

class KnownNat n => GStoreSizeSum (n :: Nat) (f :: * -> *) where gsizeSum :: f a -> Proxy n -> Int
```

the above closed type family is empy when base < 4.9; and that's a syntax error.

**Comment by phadej** [comment]
I made a revision: https://hackage.haskell.org/package/store-0.5.1.1/revisions/

Please setup a CI.

**Comment by mgsloan** [comment]
Fixed this a while back, closing.

Transferred the repo from fpco/store to mgsloan/store and setup travis CI.

**Comment by phadej** [comment]
As far as I can see, this regressed in `store-0.7`, https://matrix.hackage.haskell.org/#/package/store/0.7.0/ghc-7.10.3@1573346546

GHC-7.10

```
src/Data/Store/Internal.hs:747:10:
    Could not deduce (Generic (Complex a))
```

GHC-7.8

```
src/Data/Store/Internal.hs:746:10:
    Could not deduce (GStoreSize (GHC.Generics.Rep (Identity a)))
      arising from a use of ‘Data.Store.Impl.$gdmsize’
```

**Comment by mgsloan** [comment]
@phadej Thanks for letting me know, I believe this is fixed in 0.7.1
